### PR TITLE
[opentelemetry] Add guide to use heatmap widget with OTLP Histograms

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

Add guide on how to visualize OTLP Histograms as heatmaps.

### Motivation

There are a couple config toggles needed for this and we want to publicly document it.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by kayayarai** [comment]
Opened https://datadoghq.atlassian.net/browse/DOCS-5245 to track docs review

File: content/en/opentelemetry/guide/otlp_histogram_heatmaps.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  tag: "Documentation"
+  text: "OTLP Metric Types"
+- link: "/opentelemetry/"
+  tag: "Documentation"
+  text: "OpenTelemetry Support in Datadog"
+---
+
+## Overview
+
+The OpenTelemetry Protocol (OTLP) supports sending OTLP Histograms, a type of metric that compresses information about a set of measurements by providing aggregate statistics (sum, count, min, max...) as well as counting how many of these measurements fall into user-configurable buckets.
```
Comment (cswatt) [inline content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:15]:
```suggestion
The OpenTelemetry Protocol (OTLP) supports sending OTLP Histograms, a type of metric that compresses information about a set of measurements by providing aggregate statistics, like sum, count, min, and max. OTLP Histograms also count how many of these measurements fall into user-configurable buckets.
```

File: content/en/opentelemetry/guide/otlp_histogram_heatmaps.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+**Note**: The related OTLP Exponential Histogram type can also be displayed as a heatmap, since it is converted to a distribution. Read more about distributions on the [dedicated Distributions page][4].
+
+## Setup
+
+This guide assumes you already have a [functioning setup for sending OpenTelemetry metrics to Datadog][1].
+
+### OpenTelemetry SDK configuration
+
+If producing metrics from an OpenTelemetry SDK, you should take the following steps to configure them:
```
Comment (cswatt) [inline content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:27]:
```suggestion
If you are producing metrics from an OpenTelemetry SDK, take the following steps to configure them:
```

File: content/en/opentelemetry/guide/otlp_histogram_heatmaps.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+1. [Configure your SDK you are sending OTLP Histograms with delta temporality][2]. This will make the minimum and maximum available for the heatmap widget.
```
Comment (cswatt) [inline content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:29]:
```suggestion
1. [Configure the SDK you are sending OTLP Histograms with delta temporality][2]. This makes the minimum and maximum available for the heatmap widget.
```

File: content/en/opentelemetry/guide/otlp_histogram_heatmaps.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+2. Check if you want to override the [default bucket boundaries][3] from your aggregation. Note that each additional bucket is considered a separate custom metric.
```
Comment (cswatt) [inline content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:30]:
```suggestion
2. Check if you want to override the [default bucket boundaries][3] from your aggregation. **Note**: Each additional bucket is considered a separate custom metric.
```

File: content/en/opentelemetry/guide/otlp_histogram_heatmaps.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+- link: "/opentelemetry/"
+  tag: "Documentation"
+  text: "OpenTelemetry Support in Datadog"
+---
+
+## Overview
+
+The OpenTelemetry Protocol (OTLP) supports sending OTLP Histograms, a type of metric that compresses information about a set of measurements by providing aggregate statistics (sum, count, min, max...) as well as counting how many of these measurements fall into user-configurable buckets.
+
+You can visualize this datatype as a [heatmap][5] in Datadog by configuring your OpenTelemetry Collector Datadog exporter or Datadog Agent with the counters mode and using the 'pre-binned data' option in your heatmap.
```
Comment (cswatt) [inline content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:17]:
```suggestion
You can visualize this datatype as a [heatmap][5] in Datadog by following the steps on this page.
```

File: content/en/opentelemetry/guide/otlp_histogram_heatmaps.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+### OpenTelemetry SDK configuration
+
+If producing metrics from an OpenTelemetry SDK, you should take the following steps to configure them:
+
+1. [Configure your SDK you are sending OTLP Histograms with delta temporality][2]. This will make the minimum and maximum available for the heatmap widget.
+2. Check if you want to override the [default bucket boundaries][3] from your aggregation. Note that each additional bucket is considered a separate custom metric.
+
+For metrics coming from other sources, ensure if possible that these come as delta OTLP Histograms with the minimum and maximum fields set.
+
+### Datadog Agent or Datadog exporter configuration
```
Comment (cswatt) [inline content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:34]:
```suggestion
### Datadog Exporter or Datadog Agent configuration
```

File: content/en/opentelemetry/guide/otlp_histogram_heatmaps.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+Set the histogram mode and enable aggregation metrics on your Datadog exporter or Datadog Agent.
```
Comment (cswatt) [inline content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:36]:
```suggestion
Set the histogram mode and enable aggregation metrics on your Datadog Exporter or Datadog Agent.
```

File: content/en/opentelemetry/guide/otlp_histogram_heatmaps.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+{{< tabs >}}
+{{% tab "Datadog exporter (OTel Collector)" %}}
```
Comment (cswatt) [inline content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:39]:
```suggestion
{{% tab "Datadog Exporter (OTel Collector)" %}}
```

File: content/en/opentelemetry/guide/otlp_histogram_heatmaps.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+On the Datadog exporter section, configure the histogram mode to `counters` and enable aggregation metrics with the `send_aggregation_metrics` flag.
```
Comment (cswatt) [inline content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:41]:
```suggestion
In the `collector.yaml` file for the Datadog Exporter, configure the histogram mode to `counters` and enable aggregation metrics with the `send_aggregation_metrics` flag.
```

File: content/en/opentelemetry/guide/otlp_histogram_heatmaps.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+```yaml
+exporters:
+  datadog:
+    metrics:
+      histograms:
+        mode: counters
+        send_aggregation_metrics: true
+```
+
+**Note**: `send_aggregation_metrics` is available starting on the Datadog exporter v0.75.0. If using a previous version, use the `send_count_sum_metrics` flag instead. The minimum and maximum will be missing in previous versions.
```
Comment (cswatt) [inline content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:52]:
```suggestion
**Note**: `send_aggregation_metrics` is available starting with Datadog Exporter v0.75.0. If you are using an earlier version, use the `send_count_sum_metrics` flag instead. The minimum and maximum are missing in earlier versions.
```

File: content/en/opentelemetry/guide/otlp_histogram_heatmaps.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+```yaml
+otlp_config:
+  metrics:
+    histograms:
+      mode: counters
+      send_aggregation_metrics: true
+```
+
+**Note**: `send_aggregation_metrics` is available starting on the Datadog Agent v6.45.0/v7.45.0. If using a previous version, use the `send_count_sum_metrics` flag instead. The minimum and maximum will be missing in previous versions.
```
Comment (cswatt) [inline content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:67]:
```suggestion
**Note**: `send_aggregation_metrics` is available starting on the Datadog Agent v6.45.0/v7.45.0. If you are using an earlier version, use the `send_count_sum_metrics` flag instead. The minimum and maximum are missing in earlier versions.
```

File: content/en/opentelemetry/guide/otlp_histogram_heatmaps.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+
+{{% /tab %}}
+{{< /tabs >}}
+
+
+### Heatmap widget configuration
+
+The [heatmap widget][5] uses the set of `<YOUR METRIC NAME>.bucket` metrics generated by the Datadog Agent or Datadog exporter, each corresponding to a different histogram bucket. To visualize your histogram as a heatmap:
```
Comment (cswatt) [inline content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:75]:
```suggestion
The [heatmap widget][5] uses the set of `<YOUR METRIC NAME>.bucket` metrics generated by the Datadog Exporter or Datadog Agent, each corresponding to a different histogram bucket. To visualize your histogram as a heatmap:
```

File: content/en/opentelemetry/guide/otlp_histogram_heatmaps.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+
+1. Select `<YOUR METRIC NAME>.bucket` as the metric to visualize
```
Comment (cswatt) [inline content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:77]:
```suggestion
1. Select `<YOUR METRIC NAME>.bucket` as the metric to visualize.
```

**Review by cswatt — CHANGES_REQUESTED** [review]
requesting small edits

**Comment by mx-psi** [comment]
@cswatt mind taking another look?

**Comment by mx-psi** [comment]
@cswatt @dineshg13 PTAL!

File: content/en/opentelemetry/guide/otlp_histogram_heatmaps.md [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+The [heatmap widget][5] uses the set of `<YOUR METRIC NAME>.bucket` metrics generated by the Datadog Exporter or Datadog Agent, each corresponding to a different histogram bucket. To visualize your histogram as a heatmap:
+
+1. Select `<YOUR METRIC NAME>.bucket` as the metric to visualize.
+2. Choose the `pre-binned data` option on the `distributions of` menu.
+
+You can now see your OTLP Histogram as a heatmap widget.
+
+## OpenMetrics compatibility
+
+The [Datadog Agent OpenMetrics check][6] is also compatible with the pre-binned data heatmap widget option. If you want to send metrics to the OpenMetrics check directly without converting to OpenTelemetry, enable the `collect_histogram_buckets` and `non_cumulative_histogram_buckets` flags on your instance to ensure the data is sent in a compatible way to Datadog.
```
Comment (dineshg13) [inline content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:84]: whats the benefit of `send metrics to the OpenMetrics` ?

Comment (mx-psi) [reply content/en/opentelemetry/guide/otlp_histogram_heatmaps.md:84]: The only people I see using this are people who were already using OpenMetrics but have not yet migrated to OpenTelemetry. Datadog still supports OpenMetrics so I thought it made sense to document it, but I don't think we should encourage people to use it.

**Review by dineshg13 — APPROVED** [review]

**Review by dineshg13 — APPROVED** [review]

**Review by dineshg13 — APPROVED** [review]

**Pull request merged**